src/DataTable/DataTable.js

```diff
@@ -197,6 +197,18 @@ const DataTable = memo(({
     handleChangePage(recalculatedPage);
   }
 
+  if (paginationServer) {
```

Comment:
Please see https://reactjs.org/docs/hooks-overview.html#rules-of-hooks

> Only call Hooks at the top level. Don’t call Hooks inside loops, conditions, or nested functions.

Reply:
@janhartmann - I forgot about that and totally agree. I think we need to re-think the design the implementation of useEffect hooks in DataTable. I also think we need to move these hooks into their own files/logic.

Reply:
I thought about it as well, @jbetancur need to clean up that